app/controllers/users_controller.rb

```diff
@@ -0,0 +1,39 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :only_self_and_admin, except: :show
+
+  def show
+    # will allow you to view anyone's profile for now 
+    # some things are currently hidden in the view
+    # maybe add a separate/modified view for a public profile and restrict the dashboard?
+    @user = User.find(params[:id])
+  end
+
+  def edit
+    # only lets a user update their own profile (unless admin)
+    @user = User.find(params[:id])
+  end
+
+  def update
+    # only lets a user update their own profile (unless admin)
+    @user = User.find(params[:id])
```

Comment:
One small nitpick here is we're loading the same user from the database twice here.  Once inside the action and once in the `before_action`. My 2 cents is to move this into a helper method and use that in all views, etc.  We also would be able to remove the loading of the users from the show and edit action to.

```
helper_method :selected_user
def selected_user
  @selected_user ||= User.find_by_id(params[:id])
end
```

Then we can just call `selected_user` both in the controller and in the views and we won't have to load it explicitly multiple times.

Minor nitpick, but that's how I tend to write my code.